pkg/cmd/roachtest/cluster.go

```diff
@@ -1082,17 +1082,21 @@ func attachToExistingCluster(
 	ctx context.Context,
 	name string,
 	l *logger.Logger,
-	spec spec.ClusterSpec,
+	clusterSpec spec.ClusterSpec,
 	opt attachOpt,
 	r *clusterRegistry,
 ) (*clusterImpl, error) {
-	exp := spec.Expiration()
+	exp := clusterSpec.Expiration()
+	var attachedCloud spec.Cloud
+
 	if name == "local" {
 		exp = timeutil.Now().Add(100000 * time.Hour)
+		attachedCloud = spec.Local
```

Comment:
why do we need this edge case for local clusters?

Reply:
You're right, it's basically dead code. I just looked at the blame and assumed it was still being used but didn't bother doing reachability analysis. I'm going to remove it and simplify.

Reply:
For the record, it was removed in [1], and prior to that both `skipClusterValidationOnAttach` and `skipClusterStopOnAttach` were seeming set only by `TestRegistryRunClusterExpired`.

[1] https://github.com/cockroachdb/cockroach/pull/30977

Reply:
Ack, thanks for figuring it out. Looks like I hallucinated there being a more recent use case than 2018 😆

Reply:
Simplified. PTAL